rlberry/agents/torch/dqn/dqn.py

```diff
@@ -1,568 +1,449 @@
+    def _policy(self, observation, evaluation=False):
+        epsilon = self._epsilon_schedule(self.total_timesteps)
+        if (not evaluation) and self.rng.uniform() < epsilon:
+            if self.writer:
+                self.writer.add_scalar("epsilon", epsilon, self.total_timesteps)
+            return self.env.action_space.sample()
+        else:
+            with torch.no_grad():
+                observation = (
+                    torch.FloatTensor(observation).to(self._device).unsqueeze(0)
+                )
+                qvals_tensor = self._qnet_online(observation)[0]
+                action = qvals_tensor.argmax().item()
+                return action
 
-    def set_time(self, time):
-        self.exploration_policy.set_time(time)
+    def policy(self, observation):
+        return self._policy(observation, evaluation=True)
 
-    def eval_mode(self):
-        self.training = False
-        self.exploration_kwargs["method"] = "Greedy"
-        self.exploration_policy = exploration_factory(
-            self.env.action_space, **self.exploration_kwargs
-        )
 
-    def save(self, filename, **kwargs):
-        filename = Path(filename).with_suffix(".pickle")
-        filename.parent.mkdir(parents=True, exist_ok=True)
-        state = {
-            "state_dict": self.value_net.state_dict(),
-            "optimizer": self.optimizer.state_dict(),
-            "_writer": self._writer,
-        }
-        # torch.save(state, filename)
-        with filename.open("wb") as ff:
-            pickle.dump(state, ff)
-        return filename
-
-    @classmethod
-    def load(cls, filename, **kwargs):
-        filename = Path(filename).with_suffix(".pickle")
-        obj = cls(**kwargs)
-
-        with filename.open("rb") as ff:
-            checkpoint = pickle.load(ff)
-
-        obj.value_net.load_state_dict(checkpoint["state_dict"])
-        obj.target_net.load_state_dict(checkpoint["state_dict"])
-        obj.optimizer.load_state_dict(checkpoint["optimizer"])
-        return obj
-
-    def initialize_model(self):
-        self.value_net.reset()
-
-    def set_writer(self, writer):
-        self._writer = writer
-        try:
-            self.exploration_policy.set_writer(writer)
-        except AttributeError:
-            pass
-        if self.writer:
-            obs_shape = (
-                self.env.observation_space.shape
-                if isinstance(self.env.observation_space, spaces.Box)
-                else self.env.observation_space.spaces[0].shape
-            )
-            model_input = torch.zeros(
-                (1, *obs_shape), dtype=torch.float, device=self.device
-            )
-            self.writer.add_graph(self.value_net, input_to_model=(model_input,))
-            self.writer.add_scalar(
-                "agent/trainable_parameters", trainable_parameters(self.value_net), 0
-            )
+if __name__ == "__main__":
+    from rlberry.envs import gym_make
```

Comment:
I don't really understand why you do this. I would expect this kind of example to go into the example folder, I think that usually we don't include example in the library source. If you really want to provide an example you can put it in the docstring as an example but I think this precise example would be too ling to run (docstring examples are automatically run when the doc is constructed to check that it compiles).

Reply:
Indeed, I'm going to remove it, I was only using it to debug the implementation and forgot to delete this part!